ui/src/components/auth/RegisterContainer.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import axios from 'axios';
+import { Box, Button, Link, Stack, TextField, Typography } from '@mui/material';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { RegisterFormFields } from './helpers/form.types';
+import { formSpacing, formWidth } from './helpers/constants';
+
+// zod validation
+const registerValidationSchema = z
+  .object({
+    fullName: z.string().trim().min(1, { message: 'Please enter your full name' }).max(100),
+    email: z.string().email({ message: 'Please enter a valid email' }).trim(),
```

Comment:
actually if you will keep these validators and trimming, can keep in a helper function file